cumulus_etl/etl/tasks/nlp_task.py

```diff
@@ -0,0 +1,83 @@
+"""Base NLP task support"""
+
+import copy
+import logging
+import os
+from typing import Callable
+
+import rich.progress
+
+from cumulus_etl import common, fhir, nlp, store
+from cumulus_etl.etl.tasks.base import EtlTask, OutputTable
+
+
+class BaseNlpTask(EtlTask):
```

Comment:
This is the base NLP task I mentioned before. It basically just handles looping over the DocRefs and extracting the notes. With some boilerplate code for handling group fields if the subclass wants that.